apps/search/src/lib/algolia/algoliaUtils.ts

```diff
@@ -73,11 +74,36 @@ const mapSelectedAttributesToRecord = (attr: ProductAttributesDataFragment) => {
     return undefined;
   }
 
+  /**
+   * TODO: How/When name can be empty?
+   */
   const filteredValues = attr.values.filter((v) => !!v.name?.length);
 
+  let value: string | boolean;
+
+  /**
+   * Strategy for boolean type only, assume that only one value is possible (no option to have booleans-dropdown).
+   * REF SHOPX-332
+   * TODO: Other input types should be handled and properly mapped
+   */
+  if (
+    filteredValues.length === 1 &&
+    filteredValues[0].inputType === AttributeInputTypeEnum.Boolean &&
+    typeof filteredValues[0].boolean === "boolean"
```

Comment:
suggestion: this condition is bit complex and could be moved to separate function

Reply:
@krzysztofzuraw done